### PR TITLE
[prometheus-redis-exporter] allow using securityContext in containers

### PR DESCRIPTION
**Pull request by chesnovsky**
Signed-off-by: Stanislav Chesnovskiy <s.chesnovskiy@semrush.com>

#### What this PR does / why we need it:
This PR allows using security context in containers

#### Checklist
- [x] [DCO](https://github.com/prometheus-community/helm-charts/blob/main/CONTRIBUTING.md#sign-off-your-work) signed
- [x] Chart Version bumped
- [x] Title of the PR starts with chart name (e.g. `[prometheus-couchdb-exporter]`)

### CODE REVIEW

File: charts/prometheus-redis-exporter/templates/deployment.yaml [hunk 1]
```diff
@@ -36,6 +36,10 @@ spec:
           {{- range $key, $value := .Values.extraArgs }}
             - --{{ $key }}={{ $value }}
           {{- end }}
+          securityContext:
+          {{- range $key, $value := .Values.securityContext }}
+            {{ $key }}: {{ $value }}
+          {{- end }}
```
Comment (zanhsieh) [inline charts/prometheus-redis-exporter/templates/deployment.yaml:42]: what if securityContext in values.yaml is empty?

**Review by zanhsieh — CHANGES_REQUESTED** [review]

**Comment by acondrat** [comment]
Do you want to provide the option to configure securityContext at both POD and Container levels?

File: charts/prometheus-redis-exporter/templates/deployment.yaml (hunk 1 above)
Comment (chesnovsky) [reply charts/prometheus-redis-exporter/templates/deployment.yaml:42]: In that case, we will see empty securityContext in the manifest, and container will run under the root user by default.
I could add to the values.yaml something like this
`securityContext: {
    runAsUser: 0,
    runAsGroup: 0
}`
or this
`securityContext: {}`

**Comment by chesnovsky** [comment]
> Do you want to provide the option to configure securityContext at both POD and Container levels?

As far as I see, the POD security policy described at podsecuritypolicy.yaml has default values that could not be changed with templates.
If securityContext is in values.yaml it will not be implemented to the POD policy only to the container securityContext.

**Review by zanhsieh — APPROVED** [review]

**Pull request merged**